src/ray/core_worker/core_worker.cc

```diff
@@ -1812,7 +1799,20 @@ void CoreWorker::HandleWaitForActorOutOfScope(
   if (it == actor_handles_.end()) {
     respond(actor_id);
   } else {
-    RAY_CHECK(actor_out_of_scope_callbacks_.emplace(actor_id, std::move(respond)).second);
+    if (RayConfig::instance().gcs_actor_service_enabled()) {
```

Comment:
Can you add a comment to say why we don't need to do this if the GCS actor service is disabled?

Reply:
fixed